server/odc-service/src/main/java/com/oceanbase/odc/service/structurecompare/DefaultDBStructureComparator.java

```diff
@@ -0,0 +1,197 @@
+                tgtConfig.getConnectType().getDialectType(), tgtDbVersion);
+
+        TableStructureComparator tableComparator = new TableStructureComparator(tgtTableEditor,
+                tgtConfig.getConnectType().getDialectType(),
+                srcConfig.getSchemaName(), tgtConfig.getSchemaName());
+
+        if (srcConfig.getBlackListMap().get(DBObjectType.TABLE) == null) {
+            /**
+             * Compare all the tables between source database and target database.
+             */
+            returnVal =
+                    tableComparator.compare(new ArrayList<>(srcTableName2Table.values()),
+                            new ArrayList<>(tgtTableName2Table.values()));
+        } else {
+            /**
+             * Compare specified tables between source database and target database.
+             */
+            for (String tableName : srcConfig.getBlackListMap().get(DBObjectType.TABLE)) {
+                if (srcTableName2Table.containsKey(tableName)) {
+                    returnVal.add(tableComparator.compare(srcTableName2Table.get(tableName),
+                            tgtTableName2Table.get(tableName)));
+                } else {
+                    DBObjectComparisonResult result = new DBObjectComparisonResult(DBObjectType.TABLE, tableName,
+                            srcConfig.getSchemaName(), tgtConfig.getSchemaName());
+                    result.setComparisonResult(ComparisonResult.MISSING_IN_SOURCE);
+                    returnVal.add(result);
+                }
+            }
+        }
+        return returnVal;
+    }
+
+    private DBSchemaAccessor getDBSchemaAccessor(ConnectType connectType, DataSource dataSource, String dbVersion)
+            throws SQLException {
+        /**
+         * sysJdbcOperations and tenantName are required for OBMySQLNoGreaterThan1479SchemaAccessor to get
+         * table partition, this method will be replaced by OBMySQLGetDBTableByParser, so we just set it
+         * null here.
+         */
+        return DBSchemaAccessors.create(JdbcOperationsUtil.getJdbcOperations(dataSource.getConnection()), null,
+                connectType, dbVersion, null);
+    }
+
+    private DBTableEditor getDBTableEditor(ConnectType connectType, String dbVersion) {
+        DBObjectEditorFactory<DBTableEditor> tableEditorFactory =
+                new DBTableEditorFactory(connectType, dbVersion);
+        return tableEditorFactory.create();
+    }
+
+    private String getDBVersion(ConnectType connectType, DataSource dataSource) throws SQLException {
+        InformationExtensionPoint informationExtension = ConnectionPluginUtil.getInformationExtension(
+                connectType.getDialectType());
+        return informationExtension.getDBVersion(dataSource.getConnection());
+
+    }
+
+    private void checkConfig(DBStructureComparisonConfig srcConfig, DBStructureComparisonConfig tgtConfig) {
```

Comment:
method name should illustrate the usage, may `checkIfDialectTypeSupported`?

Reply:
rename to `checkUnsupportedConfiguration`，Because this method also checks whether the supported `DBObjectType`，t is not appropriate to change it to `checkIfDialectTypeSupported`.